torch/_refs/__init__.py

```diff
@@ -2201,17 +2210,23 @@ def _make_copy_from_view(fn):
     Given a view function (e.g. torch.diagonal) generates its copy variant (e.g. torch.diagonal_copy)
     """
     name = fn.__name__
-    fn = out_wrapper()(fn)
+    copy_name = f"{name}_copy"
+    op_copy = getattr(aten, copy_name)
 
+    @wraps(fn)
     def _fn(*args, out=None, **kwargs):
         result = fn(*args, out=out, **kwargs)
-        if out is None:
-            return result.clone(memory_format=torch.contiguous_format)
-        return result
+        if out is not None:
+            return result
 
-    copy_name = f"{name}_copy"
+        return pytree.tree_map(
+            lambda x: x.clone(memory_format=torch.contiguous_format),
+            result,
+        )
```

Comment:
all this is not needed now

Reply:
If you have issues handling multiple outputs, you should probably modify `out_wrapper` or just support those in a PR on top of this one.

Reply:
Cool idea, noted.